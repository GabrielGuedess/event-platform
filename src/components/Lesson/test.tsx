import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import { Lesson } from '.';

describe('<Lesson />', () => {
  it('should render the heading in the component', () => {
    render(
      <BrowserRouter>
        <Lesson
          key={1}
          title="Lesson 1"
          slug="lesson-1"
          availableAt={new Date()}
          lessonType="live"
        />
      </BrowserRouter>,
    );

    expect(screen.getByText(/Lesson 1/i)).toBeInTheDocument();
  });

  it('should render content status as "Em breve" based on date', () => {
    render(
      <BrowserRouter>
        <Lesson
          key={1}
          title="Lesson 1"
          slug="lesson-1"
          availableAt={
            new Date(new Date().setDate(new Date().getFullYear() - 5))
          }
          lessonType="live"
        />
      </BrowserRouter>,
    );

    expect(screen.getByText(/Em breve/i)).toBeInTheDocument();
  });

  it('should render content status as "Conteúdo liberado" based on date', () => {
    render(
      <BrowserRouter>
        <Lesson
          key={1}
          title="Lesson 1"
          slug="lesson-1"
          availableAt={new Date()}
          lessonType="live"
        />
      </BrowserRouter>,
    );

    expect(screen.getByText('Conteúdo liberado')).toBeInTheDocument();
  });

  it('should render class status as "AO VIVO" based on content type', () => {
    render(
      <BrowserRouter>
        <Lesson
          key={1}
          title="Lesson 1"
          slug="lesson-1"
          availableAt={new Date()}
          lessonType="live"
        />
      </BrowserRouter>,
    );

    expect(screen.getByText(/AO VIVO/i)).toBeInTheDocument();
  });

  it('should render class status as "AULA PRÁTICA" based on content type', () => {
    render(
      <BrowserRouter>
        <Lesson
          key={1}
          title="Lesson 1"
          slug="lesson-1"
          availableAt={new Date()}
          lessonType="class"
        />
      </BrowserRouter>,
    );

    expect(screen.getByText(/AULA PRÁTICA/i)).toBeInTheDocument();
  });
});
