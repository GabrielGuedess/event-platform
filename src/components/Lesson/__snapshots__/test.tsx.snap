// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Lesson /> should render the heading in the component 1`] = `
<a
  href="/"
>
  <span
    class="text-gray-300"
  >
    domingo • 26 de junho • 24h46
  </span>
  <div
    class="rounded border border-gray-500 p-4 mt-2"
  >
    <header
      class="flex items-center justify-between"
    >
      <span
        class="flex items-center gap-2 text-sm text-blue-500 font-medium"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 256 256"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <polyline
            fill="none"
            points="172 104 113.3 160 84 132"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <circle
            cx="128"
            cy="128"
            fill="none"
            r="96"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
        </svg>
        Conteúdo liberado
      </span>
      <span
        class="text-xs rounded px-2 py-[0.125rem] text-white border border-green-300 font-bold"
      >
        AO VIVO
      </span>
    </header>
    <strong
      class="text-gray-200 mt-5 block"
    >
      Lesson 1
    </strong>
  </div>
</a>
`;
